src/components/Signup/Signup.js

```diff
@@ -105,12 +105,16 @@ const Signup = function ({
   const [alreadyExistentAddresses, setAlreadyExistentAddresses] = useState([]);
   const [blockedDomains, setBlockedDomains] = useState([]);
 
-  const utmSource = getSource(location);
-  const utmCampaign = getParameter(location, 'utm_campaign');
-  const utmMedium = getParameter(location, 'utm_medium');
-  const utmTerm = getParameter(location, 'utm_term');
+  let utmSource, utmCampaign, utmMedium,utmTerm,utmCookies = '';
 
-  const utmCookies = manageUtmCookies(localStorage, utmSource, utmCampaign, utmMedium, utmTerm);
+  useEffect(() => {
+    utmSource = getSource(location);
+    utmCampaign = getParameter(location, 'utm_campaign');
+    utmMedium = getParameter(location, 'utm_medium');
+    utmTerm = getParameter(location, 'utm_term');
+
+    utmCookies = manageUtmCookies(localStorage, utmSource, utmCampaign, utmMedium, utmTerm);
+  }, [location]);
```

Comment:
It will generate a new entry for example if the language change.

Is it fine?

The alternative is using an empty array:

```suggestion
  }, []);
```

Reply:
Good catch! Thanks!